### PR TITLE
added selectProps to MultiSelectDropdown

### PR DESCRIPTION
**Pull request by ivoalmeida**
Adding selectProps: SelectProps as I need to set more props in this component than it is currently allowed.

### CODE REVIEW

File: packages/shared/src/dropdown/multiselectdropdown.tsx [hunk 1]
```diff
@@ -56,4 +63,5 @@ export type MultiSelectDropdownProps = {
   placeholder?: string;
   variant?: SelectVariant;
   onChange: (selected: string[]) => void;
+  selectProps?: SelectProps;
```
Comment (bipuladh) [inline packages/shared/src/dropdown/multiselectdropdown.tsx:66]: Don't we have overlap between the already existing props and `SelectProps`. Could we perhaps remove the common ones and do something like 
```
SelectProps & {
 whateverPropsAreNotPartOfSelectProps: any;
}

```

Comment (ivoalmeida) [reply packages/shared/src/dropdown/multiselectdropdown.tsx]: Done

File: packages/shared/src/dropdown/multiselectdropdown.tsx [hunk 2]
```diff
@@ -43,17 +49,17 @@ export const MultiSelectDropdown: React.FC<MultiSelectDropdownProps> = ({
...
       isCheckboxSelectionBadgeHidden
+      {...rest}
     >
       {items}
     </Select>
   );
 };
 
-export type MultiSelectDropdownProps = {
+export type MultiSelectDropdownProps = SelectProps & {
```
Comment (GowthamShanmugam) [inline packages/shared/src/dropdown/multiselectdropdown.tsx:59]: selectProps selections and placeholderText have no effect due to the selected, options.  shall we remove these custom props?

just asking for your opinion

**Review by bipuladh — APPROVED** [review]

File: packages/shared/src/dropdown/multiselectdropdown.tsx (hunk 2 above)
Comment (GowthamShanmugam) [inline packages/shared/src/dropdown/multiselectdropdown.tsx:59]: selectProps selections, placeholder text, variant, has no effect due to selected, options, and variant.  shall we remove these custom props?

just asking for your opinion 
```suggestion
 type Props = Omit<SelectProps, 'placeholderText' | 'selections'>
export type MultiSelectDropdownProps = Props & {
```

File: packages/shared/src/dropdown/multiselectdropdown.tsx [hunk 3]
```diff
@@ -1,26 +1,32 @@
...
   const [isOpen, setOpen] = React.useState(false);
 
   const onSelect = (
     _event: React.MouseEvent | React.ChangeEvent,
     selection: string
   ) => {
-    let cSelected: string[] = selected;
-    if (selected.includes(selection)) {
-      cSelected = selected.filter((item) => item !== selection);
+    let cSelected: string[] = selections as string[];
```
Comment (GowthamShanmugam) [inline packages/shared/src/dropdown/multiselectdropdown.tsx:25]:
```suggestion
    const prevSelection = selections as string[];
    let cSelected= prevSelection;
```

File: packages/shared/src/dropdown/multiselectdropdown.tsx [hunk 4]
```diff
@@ -1,26 +1,32 @@
...
+    if ((selections as string[]).includes(selection)) {
+      cSelected = (selections as string[]).filter((item) => item !== selection);
```
Comment (GowthamShanmugam) [inline packages/shared/src/dropdown/multiselectdropdown.tsx:27]:
```suggestion
    if ((prevSelection).includes(selection)) {
      cSelected = (prevSelection).filter((item) => item !== selection);
```

File: packages/shared/src/dropdown/multiselectdropdown.tsx [hunk 5]
```diff
@@ -1,26 +1,32 @@
...
     } else {
-      cSelected = [...selected, selection];
+      cSelected = [...(selections as string[]), selection];
```
Comment (GowthamShanmugam) [inline packages/shared/src/dropdown/multiselectdropdown.tsx:29]:
```suggestion
      cSelected = [...prevSelection, selection];
```

**Review by GowthamShanmugam — APPROVED** [review]

**Pull request merged**